source/common/http/http2/codec_impl.cc

```diff
@@ -208,7 +209,8 @@ void ConnectionImpl::StreamImpl::encodeTrailersBase(const HeaderMap& trailers) {
     pending_trailers_to_encode_ = createHeaderMap<HeaderMapImpl>(trailers);
   } else {
     submitTrailers(trailers);
-    parent_.sendPendingFrames();
+    auto status = parent_.sendPendingFrames();
+    RELEASE_ASSERT(status.ok(), "sendPendingFrames() failure in non dispatching context");
```

Comment:
Sadly I think this might not always be true. It's definitely possible that we can encode in the context of dispatch (like an immediate local reply). I think ultimately we are going to have to pivot on whether dispatching to decide how/when to handle errors in these functions.

Reply:
Yes, some of these methods, such as `encodeHeadersBase()` can be called when dispatching data. Note that the `sendPendingFrames()` is a noop when `dispatching_ != true` and always returns OK status.

Reply:
> Note that the sendPendingFrames() is a noop when dispatching_ != true and always returns OK status.

You mean == true, right? Since in that case we send pending frames at the end of dispatch. The problem is when we are not in dispatch. That is when I think we have to latch all errors and save them for the next dispatch.

Reply:
Oh, yes, it was a typo. `dispatching_ == true` of course.